sungc/rasterio_funcs.py

```diff
@@ -70,22 +87,25 @@ def load_bands(band_list, scale_factor, apply_scaling):
     Returns
     -------
     spectral_cube : numpy.ndarray
-        multi-band array with dimensions of [nBands, nRows, nCols]
+        multi-band array with dimensions of [nbands, nrows, ncols]
     """
-    nBands = len(band_list)
+    if scale_factor <= 0:
+        raise Exception("load_bands: scale_factor <= 0")
+
+    nbands = len(bandlist)
     spectral_cube = None
     nodata_val = None
     meta_dict = None
-    data_type = np.float32
+    data_type = np.dtype(np.float32)
```

Comment:
Not sure what this is doing ... `np.float32` will be of type `np.dtype`

Reply:
```
dt1 = np.float32
dt1.name
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'numpy.float32' has no attribute 'name'
```
However,
```
dt2 = np.dtype(np.float32)
print(dt2.name)
float32
```

Reply:
Woops my bad.  Wasn't thinking straight.